### ISSUE TITLE
Installation failing

### ISSUE THREAD

**Issue by wazxyz** [danimtb/dasshio#69]
Raspberry Pi 3B+ running latest 32bit hassio image.
The only other addons I have install are Duckdns and SSH server.

Clicking install does nothing, here is the log.
Sorry if I've done something stupid :)

19-03-11 12:53:34 INFO (SyncWorker_14) [hassio.docker.interface] Pull image danimtb/armv7-dasshio tag 0.3.1.
19-03-11 12:53:36 ERROR (SyncWorker_14) [hassio.docker.interface] Can't install danimtb/armv7-dasshio:0.3.1 -> 404 Client Error: Not Found ("pull access denied for danimtb/armv7-dasshio, repository does not exist or may require 'docker login'").

**Comment by daplaji** [comment]
Me too, I can not install. Same hardware as you.
Same error message. 
Do you have solved the issue?

**Comment by Smitty357** [comment]
Just had the same issue. I was running it before (around 2 months back, but did a fresh install last week) without any issues.
Any solution for it?

**Comment by TEO984** [comment]
Same error.
Any solution for it?

**Comment by uberhouse3** [comment]
same thing for me. first time installing and I get the same error.

**Comment by Smitty357** [comment]
Luckily I had an older install from a backed up version I could restore. But, I would like to have done a fresh install.

**Comment by dwl99** [comment]
Installation also failing for me on Raspberry Pi2 Raspbian Stretch with hass.io on Docker

**Comment by stdestro** [comment]
Same for me. Hass.io version 0.89.2 on Raspberry Pi2.
Had to restore Dasshio from backup

**Comment by daplaji** [comment]
My installation, did not work restoring to a backup. Danimtb, could you please check it?

El 17 mar. 2019 20:14, stdestro <notifications@github.com> escribió:

Same for me. Hass.io version 0.89.2 on Raspberry Pi2.
Had to restore Dasshio from backup

**Comment by eddsa** [comment]
@danimtb Same on my end. It looks like there is an issue to get the latest add-on installed. Are you able to look at it?

**Comment by crispwake** [comment]
Same error as OP here, just upgraded to 0.89.2.  Wasn’t working before or after upgrading.  Fairly fresh install.

**Comment by TheCBc** [comment]
same problem!!

**Comment by jmarlier** [comment]
idem

**Comment by jmarlier** [comment]
I managed to install Dasshio 0.0.8 by installing before Docker Enabler...

finally...doesn't work...

**Comment by gdreelin** [comment]
Same issue on Raspberry Pi 3B+ running latest 32bit hassio image. I put hthe Dasshio in the Add-On directory like some have stated but get the same deny error each time. Like it cannot find the image file.

**Comment by uberhouse3** [comment]
I finally got it to install. I migrated from pi3 b+ to NUC on Proxmox, then updated HASSio to 0.89.2. I don't know if it was just all coincidence that it finally worked after all that though.

**Comment by dwl99** [comment]
I already had a failed install with hass.io 0.89.2. Have just updated to 0.90.0 and still not installing.

**Comment by yanniboi** [comment]
OK, this is happening for me too.

Running hassio 0.89.2 on RPi 3b

Couldn't find anything is HA error log so tried to install via the cli:

`hassio addons install 03d23610_dasshio --log-level debug --raw-json `

```
DEBU[0000] Adding homedir to searchpath                  homedir=/root
INFO[0000] No configfile found                          
DEBU[0000] Debug flags                                   apiToken=[API TOKEN] cfgFile= endpoint=hassio logLevel=debug rawJSON=true
DEBU[0000] addons install                                args="[03d23610_dasshio]"
DEBU[0000] [GenerateURI]                                 base=hassio command="{slug}/install" section=addons
DEBU[0000] [GenerateURI] Result                          uri="http://hassio/addons/{slug}/install" url="http://hassio/addons/%7Bslug%7D/install" url(string)="http://hassio/addons/{slug}/install"

{"result": "error", "message": null}

```

Super helpful :face_with_head_bandage: 

So I forked the repo to trigger travis to re-run:

https://travis-ci.org/yanniboi/dasshio/builds/509915201

Starting to get somewhere:
`No Gemfile found, skipping bundle install` 

Could this be something? Is there anything in the docker image or the requirements.txt that is out of date?

**Comment by yanniboi** [comment]
Nope I was wrong, the repo travis build just has a load of secure environment variables which mean I don't even know what docker image you are testing in :/ sorry cant help debug any further for now then.

**Comment by yanniboi** [comment]
Ok! Problem found!

The issue was in the system log the whole time.

```
19-03-22 14:24:15 ERROR (SyncWorker_15) [hassio.docker.interface] Can't install danimtb/armv7-dasshio:0.3.1 -> 404 Client Error: Not Found ("pull access denied for danimtb/armv7-dasshio, repository does not exist or may require 'docker login'").
```

Had a quick look on docker hub to see what the problem is (https://hub.docker.com/u/danimtb) and there you go. There is no docker image for the armv7 arch. Must be new...

![image](https://user-images.githubusercontent.com/1744957/54830385-51b0b100-4cb0-11e9-8182-8ce1a980b847.png)

So the proper solution is simply for @danimtb to add a new docker image for the new architecture (shouldn't be too hard?)

In the mean time I have hacked away the repo to force using amd64 and put that in a PR. I tried it locally on my HA instance and it works now :smile_cat: 

https://github.com/danimtb/dasshio/pull/70/files

Hope this helps...

**Comment by gdreelin** [comment]
Will the amd64 work on Pi 3 B+? or should I change it to another one? Now strangely enough it sort of loaded late last night but it seem to half load only not sure what is going on there.

**Comment by vincegre** [comment]
@danimtb Any chance to get that fixed soon ? or should we find an other solution and this project dead ? Thanks

**Comment by bahuma20** [comment]
Hey @danimtb :wave: 

Could you please look at the comment from @yanniboi (https://github.com/danimtb/dasshio/issues/69#issuecomment-475647045). It contains the concrete problem and how to solve it.

We all would really appreciate if you could investigate in this. 

Thank you in advance :)

**Comment by dwl99** [comment]
Would it work if someone uploaded a backup of just this package and we restored from that?

**Comment by gdreelin** [comment]
I was able to load it after I copied it over to addons directory but I think there is something still wrong with it. I starts then stops with a fail error no code. Not really sure which version to use on a Rasp Pi so it was guess work on my part. There is armhf, aacrch64, amd64, and i386. I know not to use the i386 on a pi but not sure about the other three, new with pi's.

**Comment by cheechie** [comment]
I am assuming that this is not going to be fixed, since the developer has not responded?

**Comment by vincegre** [comment]
Isn't it possible to fork the project and make the correction in it to get it working again ?

**Comment by jmarlier** [comment]
I have a raspberry pi 3 B with hassio. I managed to install Dasshio locally in folder addons, because I had to modify the file config.json so that it takes the good model of processor which is an "armhf"

Here are the changes I added 👍 


  "url": "https://github.com/danimtb/dasshio",
  "startup": "services",
  _"arch": [
    "armhf"
  ],_
  "boot": "auto",

Thanks to yanniboi who allowed me to find the solution.

**Comment by dwl99** [comment]
> 
> 
> I have a raspberry pi 3 B with hassio. I managed to install Dasshio locally in folder addons, because I had to modify the file config.json so that it takes the good model of processor which is an "armhf"
> 
> Here are the changes I added 👍
> 
> "url": "https://github.com/danimtb/dasshio",
> "startup": "services",
> _"arch": [ "armhf" ],_
> "boot": "auto",
> 
> Thanks to yanniboi who allowed me to find the solution.

Are you running hassio in Docker? 
What exactly did you copy to the add-ons folder? Was it the entire dasshio folder you copied to `/usr/share/hassio/addons/core` ?
Can you share the entire contents of your config.json with the right formatting?

Thanks!

**Comment by jmarlier** [comment]
I connect with Samba to directly access the Addons directory. I can not say more. i'm self taught.

dwl99 what do you use? Rasbian, hassbian, hassos ??

**Comment by vincegre** [comment]
@forast3r made a fork working right now with latest requirements of HA: https://github.com/forast3r/dasshio
You can also just download the archive from here, extract the sub-folder dasshio and copy it in addons directory of your hassio system. Just edit the config file (with a pure text editor such as vim/nano whatever) to add the arch field that is mandatory now with HA and so it should look like that. Just find out at begin of file that section as listed below and add the 3 lines extra as below starting with "arch.... Best is to do copy/paste to avoid any mistakes... amd64 is for the ones running hassio on Intel NUC like me, if on Raspberry you should put armhf ;) enjoy !

```
"url": "https://github.com/danimtb/dasshio",
  "startup": "services",
  "arch": [
  "amd64"
  ],
  "boot": "auto",
  "host_network": true,
```

**Comment by dwl99** [comment]
> 
> 
> I connect with Samba to directly access the Addons directory. I can not say more. i'm self taught.
> 
> dwl99 what do you use? Rasbian, hassbian, hassos ??

Thanks for this, I'm on Raspbian Stretch with a Dockerised hass.io

My config.json file looks like this
```
{
  "addons_custom_list": [
    "https://github.com/hassio-addons/repository",
    "https://github.com/forast3r/dasshio.git"
  ],
  "timezone": "Europe/London",
  "wait_boot": 5,
  "last_boot": "2019-04-08T17:46:07"
}
```
I'm a little unclear which lines to add & where if someone wouldn't mind helping a HA noob :-)

**Comment by gdreelin** [comment]
> 
> 
> @forast3r made a fork working right now with latest requirements of HA: https://github.com/forast3r/dasshio
> You can also just download the archive from here, extract the sub-folder dasshio and copy it in addons directory of your hassio system. Just edit the config file (with a pure text editor such as vim/nano whatever) to add the arch field that is mandatory now with HA and so it should look like that. Just find out at begin of file that section as listed below and add the 3 lines extra as below starting with "arch.... Best is to do copy/paste to avoid any mistakes... amd64 is for the ones running hassio on Intel NUC like me, if on Raspberry you should put armhf ;) enjoy !
> 
> ```
> "url": "https://github.com/danimtb/dasshio",
>   "startup": "services",
>   "arch": [
>   "amd64"
>   ],
>   "boot": "auto",
>   "host_network": true,
> ```

Worked great from you repository!! Now just have to get it to work on HA...

**Comment by jmarlier** [comment]
@dwl99 I was talking about the config.json of the Dasshio file
if you are on Raspbian Strech you can install this : https://github.com/JulianKahnert/amazon-dashbutton 
When you connect to the folder by Samba, what do you have?
Dasshio is only for Hassio but you can install it differently
excuse my english level...I am better in IT than in language

**Comment by dwl99** [comment]
> 
> 
> @dwl99 I was talking about the config.json of the Dasshio file
> if you are on Raspbian Strech you can install this : https://github.com/JulianKahnert/amazon-dashbutton
> When you connect to the folder by Samba, what do you have?
> Dasshio is only for Hassio but you can install it differently
> excuse my english level...I am better in IT than in language

Ah, thanks, I was looking at the wrong config.json file! I am running hassio in a Docker container so I think I'll persevere with dasshio!

**Comment by dwl99** [comment]
I'm stuck! I have hassio running in a Docker container on Raspbian Stretch (RPi2). I copied the dasshio folder (contains 4 files) to what I think is the add-ons folder at `/usr/share/hassio/addons/core`

I edited the config.json file in the dashio folder and it now looks like this
```
{
  "name": "Dasshio",
  "version": "0.3.1",
  "arch": ["amd64"],
  "slug": "dasshio",
  "description": "Use Amazon Dash Buttons in Home Assistant",
  "url": "https://github.com/danimtb/dasshio",
  "startup": "services",
    "arch": [
    "armhf"
  ],
  "boot": "auto",
  "host_network": true,
  "homeassistant_api": true,
  "image": "danimtb/{arch}-dasshio",
  "options": {
    "timeout": 20,
    "buttons": [
      {
        "name": "null",
        "address": "null",
        "domain": "",
        "service": "",
        "service_data": ""
      }
    ]
  },
  "schema": {
    "timeout": "int",
    "buttons": [
      {
        "name": "str",
        "address": "str",
        "url": "str?",
        "headers": "str?",
        "body": "str?",
        "domain": "str?",
        "service": "str?",
        "service_data": "str?"
      }
    ]
  }
}
```
But dasshio is nowhere to be seen in the hassio dashboard. Where have I gone wrong?

Thanks

**Comment by forast3r** [comment]
@dwl99 As stated above, you should change ["amd64"] with "["armhf"] so your system grabs the right image for your controller architecture.

**Comment by jmarlier** [comment]
@dwl99 I don't know hassio running in Docker / try to copy the addon to folder /usr/share/hassio/addons (without core)

**Comment by danimtb** [comment]
Fixed by #74

**Comment by bahuma20** [comment]
@danimtb Thank you very much. Works like a charm 👏 👍
